crates/rune/src/runtime/unit.rs

```diff
@@ -35,10 +36,32 @@ pub type DefaultStorage = ArrayUnit;
 #[cfg(rune_byte_code)]
 pub type DefaultStorage = ByteCodeUnit;
 
-/// Instructions from a single source file.
+/// Instructions and debug info from a single source file.
 #[derive(Debug, Clone, Default, Serialize, Deserialize)]
 #[serde(bound = "S: Serialize + DeserializeOwned")]
 pub struct Unit<S = DefaultStorage> {
+    /// Instructions.
+    #[serde(flatten)]
+    data: UnitData<S>,
+    /// Debug info if available for unit.
+    debug: Option<Box<DebugInfo>>,
+}
+impl<S> Deref for Unit<S> {
```

Comment:
While I see that this minimizes the amount of lines changed. I'd prefer not to have the deref implementation. So something like`self.storage` would become `self.data.storage` below.

Reply:
Yeah, I was going to do that initially, but wasn't sure which you'd prefer for maintainability.